pkg/object/httpserver/mux.go

```diff
@@ -544,7 +569,7 @@ func (mi *muxInstance) serveHTTP(stdw http.ResponseWriter, stdr *http.Request) {
 }
 
 func (mi *muxInstance) search(req *httpprot.Request) *route {
-	headerMismatch, methodMismatch := false, false
+	headerMismatch, methodMismatch, queryMismatch := false, false, false
```

Comment:
the comment at L576 - L578 should also be updated to mention that we are sure the cached path does not contain any queries.